sim/src/tlv.rs

```diff
@@ -317,12 +331,17 @@ impl ManifestGen for TlvGen {
     }
 
     fn protect_size(&self) -> u16 {
-        if self.dependencies.is_empty() {
-            0
-        } else {
-            // Include the header and space for each dependency.
-            4 + (self.dependencies.len() as u16) * (4 + 4 + 8)
+        let mut size = 0;
+        if !self.dependencies.is_empty() || (Caps::HwRollbackProtection.present() && self.security_cnt.is_some()) {
+            // include the header.
+            size += 4;
+            // add space for each dependency.
+            size +=  (self.dependencies.len() as u16) * (4 + 4 + 8);
```

Comment:
Size of a Dependency struct instead? I know it was (4+4+8) before.

Reply:
I've changed it, but a +4 is still required.

Reply:
ACK.